core/app/models/spree/credit_card.rb

```diff
@@ -66,11 +66,10 @@ def expiry=(expiry)
     #
     # @param num [String] the desired credit card number
     def number=(num)
-      @number = begin
-                  num.gsub(/[^0-9]/, '')
-                rescue
-                  nil
-                end
+      @number =
+        if num.is_a?(String)
```

Comment:
Let's discuss this elsewhere. I would like to follow this up and replace this method with something that more resembles a standard AR string attribute (that will cast Integers to a string), but I felt that that was too big of a change here.